packages/wrangler/pages/functions/template-worker.ts

```diff
@@ -110,29 +110,23 @@ export default {
       }
 
       const { value } = handlerIterator.next();
-      if (value) {
```

Comment:
There is no need to check the value (according to the types) since it will always be defined.

Reply:
Always present, but not always truthy. Once the iterator is exhausted, `value` will be `undefined`.

Reply:
Doh! I should have remembered.

So the "proper" way should be:

```ts
const {value, done} = handleIterator.next();
if (!done) {
 ...
}
```

But then that would mean that we don't handle the final `return` statement from the `executeRequest()` generator.

So I can see that this is the most appropriate approach. Thanks for catching this.
(If only we had unit tests for this...)

Reply:
Oh actually, I remember why we needed to remove this.
If we have the `if (value)` then we must also have an `else` clause that returns "something". Otherwise the return type of `next()` becomes `Promise<Response|undefined>` which is not compatible with the `EventContext` type.

Reply:
@GregBrimble - can you take a new look at this file now? I have refactored the code a bit.